### ISSUE TITLE
Bug: optimizer wrongly removes temporary variable

### ISSUE THREAD

**Issue by FilippoPolo** [intellifactory/websharper#695]
The following code:

```
type testtype = { mutable r: int }
let test_ws_bug t =
  let old = t.r in
  t.r <- 1;
  old
```

Gets translated to:

```
 test_ws_bug=function(t)
 {
  t.r=1;
  return t.r;
 };
```

Which is wrong. The original code returns the previous value of the mutable, while the translated code will always return 1. That temp variable "old" cannot just be replaced with t.r, because t.r is mutable.

Tested in Zafir 4.0.160.40-beta6.

**Comment by Jand42** [comment]
Thanks for the report, quick fix coming.

**Comment by Jand42** [comment]
Fix released in version `4.0.161.45-beta6`
